x-pack/plugins/security_solution/public/management/pages/policy/view/policy_settings_form/components/protection_setting_card_switch.tsx

```diff
@@ -82,6 +82,11 @@ export const ProtectionSettingCardSwitch = React.memo(
                 newPayload[os].popup[protection as LinuxPolicyProtection].enabled =
                   event.target.checked;
               }
+              if (protection === 'behavior_protection') {
+                newPayload.windows.behavior_protection.reputation_service = event.target.checked;
+                newPayload.mac.behavior_protection.reputation_service = event.target.checked;
+                newPayload.linux.behavior_protection.reputation_service = event.target.checked;
+              }
```

Comment:
I agree that this code could use a refactor, however, I believe this should be a separate ticket to tackle all these places at the same time.

Reply:
Changed

Reply:
I see, you are right!

I wasn't aware of the windows only thing - but with this in mind, the `if (os === 'windows') { ...` approach makes sense, and probably your code should use it, too, so it won't enable or disable `reputation_service` for other OSs than the ones received in `osList`

Reply:
The reputation service is a part of the behavior protection card only, which applies to all three OSs ;)

Reply:
I see your point. still, the `if`s you added are performed 3 times in a row unnecessarily, so I think they should be moved out of the `for` loop (and even the `if` checking the target?) at least.

although, I think that the safest bet is to use the same os-based `if` approach, in case there is an environment where behaviour protection is not applied to all OSs - just to keep the logic more cohesive and less prone to future bugs. what do you think?